### PR TITLE
feat: show and hide solutions

### PR DESCRIPTION
**Pull request by TomWoodward**

### CODE REVIEW

**Review by philschatz — CHANGES_REQUESTED** [review]
Ah, here is [an example](https://rex-web-pr-303.herokuapp.com/books/college-physics/pages/2-2-vectors-scalars-and-coordinate-systems#fs-id1788025)

It seems to take a while to hide the element, is that intentional?

Kapture 2019-09-10 at 17 06 24

It does not seem to expand when search results find text in it. Should it?

Kapture 2019-09-10 at 17 18 55


Is @Malar-Natarajan still testing this with a screenreader and with search results?

*Aside:* I was not able to click the search icon when my screen was narrow; is that desired behavior?

*Aside:* It seems _really hard_ to clear the search results. I clicked, entered `speed scalar magnitude`, clicked the `(X)`, reloaded the page, and clicked in the input box and viola! my previous search was back! (see [screencast](https://user-images.githubusercontent.com/253202/64655952-69612a80-d3f3-11e9-9c0c-f0561b7497af.gif))

File: src/app/content/components/Page.tsx [hunk 1]
```diff
@@ -248,23 +263,57 @@ export class PageComponent extends Component<PropTypes> {
...
+  private toggleSolution = (button: HTMLElement) => () => {
+    if (!button.parentElement || !button.parentElement.parentElement) {
+      return;
+    }
+    const solution = button.parentElement.parentElement;
+
+    if (solution.matches('.ui-solution-visible')) {
+      solution.className = solution.className.replace('ui-solution-visible', '');
+      solution.removeAttribute('aria-expanded');
+      solution.setAttribute('aria-label', 'show solution');
```
[the remark this answers is not in the document]
Comment (TomWoodward) [reply src/app/content/components/Page.tsx]: good call

File: src/app/content/components/Page.tsx [hunk 2]
```diff
@@ -234,7 +236,20 @@ export class PageComponent extends Component<PropTypes> {
...
     });
-}
+  }
+
+  private wrapSolutions(rootEl: HTMLElement) {
+    // Wrap solutions in a div so "Show/Hide Solutions" work
+    rootEl.querySelectorAll('.exercise .solution, [data-type="exercise"] [data-type="solution"]').forEach((el) => {
+      const contents = el.innerHTML;
+      el.innerHTML = `
+        <div class="ui-toggle-wrapper">
```
[the remark this answers is not in the document]
Comment (TomWoodward) [reply src/app/content/components/Page.tsx]: i copied what was already being used on cnx

**Comment by TomWoodward** [comment]
*It seems to take a while to hide the element, is that intentional?*
its not ideal but we'd need to prerender in the modifications to make this better and this one is complicated enough that i'd want to change the prerendered modifications to run in a dom processor instead of just regex. so i talked with kim about it and we're accepting this behavior for now.

*It does not seem to expand when search results find text in it. Should it?*
malar found this issue, kim decided to make a new card for it.

*Aside: I was not able to click the search icon when my screen was narrow; is that desired behavior?*
i don't recognize this issue, can you give some more details?

*Aside: It seems really hard to clear the search results. I clicked, entered speed scalar magnitude, clicked the (X), reloaded the page, and clicked in the input box and viola! my previous search was back!*
#307  fixes this

**Review by philschatz — APPROVED** [review]
Yay!

**Pull request merged**